sheet_functions/general.js

```diff
@@ -151,39 +209,44 @@ function TERNARY_TO_NUMBER(ternaryString) {
 /**
  * Given a string of morse code, convert it to plain text.
  *
- * @param {String} the input text convert
- * @param {String} the optional string to use for a dot
- * @param {String} the optional string to use for a dash
+ * @param {string} input the input text to convert
+ * @param {string=} dotChar the symbol to use for a dot (optional, default=".")
+ * @param {string=} dashChar the symbol to use for a dash (optional, default="-")
+ * @return {string} the decoded string
+ *
  * @customfunction
  */
-function FROM_MORSE(input, optDotChar, optDashChar) {
+function FROM_MORSE(input, dotChar, dashChar) {
   return _forEachWord(input, function(word) {
     var normalizedString = new String(word);
 
     // convert word to dots and dashes
-    if (optDotChar != undefined || optDotChar != null) {
-      normalizedString = _gsub(normalizedString, optDotChar, '.');
+    if (dotChar != undefined || dotChar != null) {
+      normalizedString = _gsub(normalizedString, dotChar, '.');
     }
 
-    if (optDashChar != undefined || optDashChar != null) {
-      normalizedString = _gsub(normalizedString, optDashChar, '-');
+    if (dashChar != undefined || dashChar != null) {
```

Comment:
fixed, this actually had a bug before that was always evaluating the inside.  everything is either not-undefined or not-null!